src/util.h

```diff
@@ -258,14 +262,41 @@ inline void MakeTmpdir(std::string* path) {
   *path = std::string(tmppath_buffer) + "gflags_unittest";
   _mkdir(path->c_str());
 }
-#else
+#elif defined(HAVE_SYS_STAT_H)
 inline void MakeTmpdir(std::string* path) {
   if (!path->empty()) {
 	int err = mkdir(path->c_str(), 0755);
 	if (err == 0 || errno == EEXIST) return;
   }
   mkdir("/tmp/gflags_unittest", 0755);
 }
+#elif __cplusplus >= 201703L
+inline void MakeTmpdir(std::string* path) {
```

Comment:
Would you mind explaining the reason for these two variants using `mkdtemp`? I prefer to keep only one.